mockwebserver/src/main/java/okhttp3/mockwebserver/RecordedRequest.kt

```diff
@@ -14,154 +14,102 @@
-      InetAddress inetAddress = socket.getLocalAddress();
+      val scheme = if (socket is SSLSocket) "https" else "http"
+      val inetAddress = socket.localAddress
 
-      String hostname = inetAddress.getHostName();
-      if (inetAddress instanceof Inet6Address) {
-        hostname = "[" + hostname + "]";
+      var hostname = inetAddress.hostName
+      if (inetAddress is Inet6Address) {
+        hostname = "[$hostname]"
       }
 
-      int localPort = socket.getLocalPort();
+      val localPort = socket.localPort
       // Allow null in failure case to allow for testing bad requests
       this.requestUrl =
-          HttpUrl.parse(String.format("%s://%s:%s%s", scheme, hostname, localPort, path));
+          HttpUrl.parse(String.format("%s://%s:%s%s", scheme, hostname, localPort, path))
     } else {
-      this.requestUrl = null;
-      this.method = null;
-      this.path = null;
+      this.requestUrl = null
+      this.method = null
+      this.path = null
     }
   }
 
-  public HttpUrl getRequestUrl() {
-    return requestUrl;
-  }
-
-  public String getRequestLine() {
-    return requestLine;
-  }
-
-  public String getMethod() {
-    return method;
-  }
-
-  public String getPath() {
-    return path;
-  }
-
-  /** Returns all headers. */
-  public Headers getHeaders() {
-    return headers;
-  }
-
-  /** Returns the first header named {@code name}, or null if no such header exists. */
-  public String getHeader(String name) {
-    List<String> values = headers.values(name);
-    return values.isEmpty() ? null : values.get(0);
-  }
-
-  /**
-   * Returns the sizes of the chunks of this request's body, or an empty list if the request's body
-   * was empty or unchunked.
-   */
-  public List<Integer> getChunkSizes() {
-    return chunkSizes;
-  }
-
-  /**
-   * Returns the total size of the body of this POST request (before truncation).
-   */
-  public long getBodySize() {
-    return bodySize;
-  }
-
-  /** Returns the body of this POST request. This may be truncated. */
-  public Buffer getBody() {
-    return body;
-  }
-
-  /** @deprecated Use {@link #getBody() getBody().readUtf8()}. */
-  public String getUtf8Body() {
-    return getBody().readUtf8();
-  }
-
-  /**
-   * Returns the index of this request on its HTTP connection. Since a single HTTP connection may
-   * serve multiple requests, each request is assigned its own sequence number.
-   */
-  public int getSequenceNumber() {
-    return sequenceNumber;
-  }
-
-  /** Returns the connection's TLS version or null if the connection doesn't use SSL. */
-  public TlsVersion getTlsVersion() {
-    return handshake != null ? handshake.tlsVersion() : null;
-  }
-
-  /**
-   * Returns the TLS handshake of the connection that carried this request, or null if the request
-   * was received without TLS.
-   */
-  public Handshake getHandshake() {
-    return handshake;
+  /** Returns the first header named `name`, or null if no such header exists.  */
```

Comment:
`Returns the first header named [name],...` in square braces, not backticks

Reply:
Kotlin lets you reference parameters; Java doesn’t so we have to use crude `{@code }` blocks

Reply:
Suddenly see the light,Fixed.